### ISSUE TITLE
Any sense to make it work in nodejs?

### ISSUE THREAD

**Issue by dy** [chrisguttandin/worker-timers#358]
Would be useful to have this library in node too, isn't that?
There's for example https://www.npmjs.com/package/webworker-threads.

Or possibly - just a stub with `setTimeout` fallback

**Comment by chrisguttandin** [comment]
Hi @dy, there is even a native package now in the core of Node.js called [`worker_threads`](https://nodejs.org/dist/latest-v12.x/docs/api/worker_threads.html). However I still think it wouldn't make sense to have a Node.js implementation. `worker-timers` is meant to solve a problem which only exists in the browser and therefore I still believe it would cause confusion if it would work in Node.js.

`worker-timers` is useful to avoid timer throttling in websites that loose focus. As far as I know there is no comparable problem in Node.js. Please let me know if there is some comparable behavior that I don't know of.

**Comment by dy** [comment]
I mean, just from consumer perspective - some modules are cross-platform,
and having stable cross-platform timers dependency is something normal to
look for. That’s fine if node timers are already stable.
Otherwise that requires wrappers such as
https://github.com/unihooks/unihooks/blob/master/src/util/timers.js
But I guess that indeed can be concern of a separate module, just thought
that’d be easier to have simple node entry here.

**Comment by chrisguttandin** [comment]
Hi @dy, I think you're doing it exactly right. The technique is described in the [Server-Side Rendering](https://github.com/chrisguttandin/worker-timers#server-side-rendering) paragraph of the readme.

To make this a little easier I thought about instantiating the worker lazily in all my modules which use a Web Worker ([chrisguttandin/web-audio-beat-detector#285](https://github.com/chrisguttandin/web-audio-beat-detector/issues/285#issuecomment-538027934)). It would then be possible to import `worker-timers` in Node.js code but it wouldn't be possible to use it. You could remove the file replacement and instead do something like this:

```js
import * as workerTimers from 'worker-timers;

export const setInterval = (isBrowser) ? workerTimers.setInterval : global.setInterval;
```

Would that be helpful?

**Comment by dy** [comment]
@chrisguttandin yes, pretty much, that's significantly easier.

**Comment by chrisguttandin** [comment]
Hi @dy, the Web Worker gets instantiated lazily now as proposed above. I published the update as v5.1.0.
